sa/sa.go

```diff
@@ -1513,7 +1513,7 @@ func (ssa *SQLStorageAuthority) SetOrderError(ctx context.Context, order *corepb
 
 	om, err := orderToModel(order)
```

Comment:
Same here, we are opening a transaction then doing unrelated work, so if `orderToModel` fails we've opened then rolled back a transaction for no real reason, seems like opening should be deferred until we are actually ready to do queries.